src/Microsoft.EntityFrameworkCore.Tools.DotNet/Internal/ProjectContextFactory.cs

```diff
@@ -14,19 +16,57 @@ public class ProjectContextFactory
     {
         public IProjectContext Create(string filePath,
             NuGetFramework framework,
-            string configuration = null,
-            string outputDir = null)
+            string configuration,
+            string outputDir)
+        {
+            configuration = configuration ?? Constants.DefaultConfiguration;
+
+            if (Directory.Exists(filePath))
+            {
+                var msbuildFiles = Directory.EnumerateFiles(filePath, "*.*proj").Take(2).ToList();
```

Comment:
This will match xproj too. Probably better to filter that since it's not actually an "MSBuild" project.